base/dcp_dest.go

```diff
@@ -0,0 +1,406 @@
+package base
+
+import (
+	"context"
+	"errors"
+	"expvar"
+	"fmt"
+	"io"
+	"strconv"
+
+	"github.com/couchbase/cbgt"
+	"github.com/couchbase/gomemcached"
+	sgbucket "github.com/couchbase/sg-bucket"
+	pkgerrors "github.com/pkg/errors"
+)
+
+// vbucketIdStrings is a memorized array of 1024 entries for fast
+// conversion of vbucketId's to partition strings via an index lookup.
+// (Atoi is faster than map lookup when going in the other direction)
+var vbucketIdStrings []string
```

Comment:
Could use an array here given it's fixed length for the purposes of lookups, and then drop the make() inside init.

```suggestion
var vbucketIdStrings [1024]string
```